src/agent/border_agent.cpp

```diff
@@ -81,89 +84,91 @@ enum
     kBorderAgentUdpPort = 49191, ///< Thread commissioning port.
 };
 
-BorderAgent::BorderAgent(Ncp::Controller *aNcp)
+uint32_t BorderAgent::StateBitmap::ToUint32(void) const
+{
+    uint32_t bitmap = 0;
+
+    bitmap |= mConnectionMode & (0x07 << 0);
+    bitmap |= mThreadIfStatus & (0x03 << 3);
+    bitmap |= mAvailability & (0x03 << 5);
+    bitmap |= mBbrIsActive & (0x01 << 7);
+    bitmap |= mBbrIsPrimary & (0x01 << 8);
```

Comment:
Is the order of operations correct here? It seems we are shifting the mask first and then applying bit-wise & operator?

Reply:
@jwhui Thanks! You are right and I think we actually don't need to bit-wise & since those member fields are already bitfields. I updated, PTAL.